### PR TITLE
Set uncertainty values to null if they are codes

### PR DESCRIPTION
**Pull request by mayerm94**
Uncertainty low and high were being set to codes, causing codes to be compared to enumerables in instances where a value could be either a code or quantity (QDM.Result)

Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
- [ ] All dependent libraries are appropriately updated or have a corresponding PR related to this change

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer:**

Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: src/datatypes/uncertainty.coffee [hunk 1]
```diff
@@ -11,6 +11,10 @@ module.exports.Uncertainty = class Uncertainty
         # Uncertainties with different types probably shouldn't be supported.
         return false
       if typeof a.after is 'function' then a.after b else a > b
+    isCode = (val) ->
```
Comment (losborne) [inline src/datatypes/uncertainty.coffee]: I think we should follow the recent pattern in `isQuantity` or `isDatetime` and add this to `src/datatypes/clinical.coffee`:

```coffeescript
  # Define a simple getter to allow type-checking of this class without instanceof
  # and in a way that survives minification (as opposed to checking constructor.name)
  Object.defineProperties @prototype,
    isCode:
      get: -> true
```

Then this logic would change to:
```coffeescript
if @low.isCode then @low = null
if @high.isCode then @high = null
```

Comment (cmoesel) [reply src/datatypes/uncertainty.coffee]: @losborne makes a great point -- and I don't think it should be hard to implement (given he more or less included the implementation in his comment).  So... I'd suggest we follow his advice.

Comment (cmoesel) [reply src/datatypes/uncertainty.coffee]: @mayerm94 mentioned in a private thread that we might also need to do the same for `ValueSet` and `Concept`.  I'm good with that as well!

File: src/datatypes/uncertainty.coffee [hunk 2]
```diff
@@ -11,7 +11,11 @@ module.exports.Uncertainty = class Uncertainty
         # Uncertainties with different types probably shouldn't be supported.
         return false
       if typeof a.after is 'function' then a.after b else a > b
+    isNonEnumerable = (val) ->
+      val? and (val.isCode or val.isConcept or val.isValueSet)
     if typeof @high is 'undefined' then @high = @low
+    if isNonEnumerable(@low) then @low = null
+    if isNonEnumerable(@high) then @high = null
```
Comment (cmoesel) [inline src/datatypes/uncertainty.coffee]: I hadn't thought of this before, but your unit tests made me think of it.  This could end up generating something like: `Uncertainty(1, null)`.  This concerns me because in the `lessThan` implementation (which everything else drives off of), `null` appears to have special meaning (it seems to be neg/pos infinity):
https://github.com/cqframework/cql-execution/blob/19775fd74c97c5d054e69f5aa6a050d9bb4fe450/src/datatypes/uncertainty.coffee#L48-L55

We don't want a code to be misinterpreted as an infinity.  I think it might be better that if either `@low` or `@high` are not enumerable, then we set _both_ to `null`.  In that case, I think `lessThan` will always return null.

What do you think of that?
